core/src/main/scala/com/nvidia/spark/rapids/tool/Platform.scala

```diff
@@ -0,0 +1,157 @@
+ */
+class Platform(platformType: String) {
+  /**
+   * Recommendations to be excluded from the list of recommendations.
+   * These have the highest priority.
+   */
+  val recommendationsToExclude: Seq[String] = Seq.empty
+  /**
+   * Recommendations to be included in the final list of recommendations.
+   * These properties should be specific to the platform and not general Spark properties.
+   * For example: "spark.databricks.optimizer.dynamicFilePruning" for the Databricks platform.
+   *
+   * Represented as a tuple of (propertyKey, propertyValue).
+   */
+  val recommendationsToInclude: Seq[(String, String)] = Seq.empty
+  /**
+   * Dynamically calculates the recommendation for a specific Spark property by invoking
+   * the appropriate function based on `sparkProperty`.
+   * TODO: Implement this function and integrate with existing code in AutoTuner
+   *
+   * @param sparkProperty The Spark property for which the recommendation is calculated.
+   * @param args Variable list of arguments passed to the calculation function for dynamic
+   *             processing.
+   * @return Optional string containing the recommendation, or `None` if unavailable.
+   */
+  def getRecommendation(sparkProperty: String, args: Any*): Option[String] = None
+
+  /**
+   * Checks if the `property` is valid:
+   * 1. It should not be in exclusion list
+   *   OR
+   * 2. It should be in the inclusion list
+   */
+  def isValidRecommendation(property: String): Boolean = {
+    !recommendationsToExclude.contains(property) ||
+      recommendationsToInclude.map(_._1).contains(property)
+  }
+
+  /**
+   * Checks if the `comment` is valid:
+   * 1. It should not have any property from the exclusion list
+   */
+  def isValidComment(comment: String): Boolean = {
+    recommendationsToExclude.forall(excluded => !comment.contains(excluded))
+  }
+
+  def getOperatorScoreFile: String = {
+    s"operatorsScore-$platformType.csv"
+  }
+}
+
+class DatabricksPlatform(platformType: String) extends Platform(platformType) {
+  override val recommendationsToExclude: Seq[String] = Seq(
+    "spark.executor.cores",
+    "spark.executor.instances",
+    "spark.executor.memory",
+    "spark.executor.memoryOverhead"
+  )
+  override val recommendationsToInclude: Seq[(String, String)] = Seq(
+    ("spark.databricks.optimizer.dynamicFilePruning", "false")
+  )
+}
+
+class DataprocPlatform(platformType: String) extends Platform(platformType)
+
+class EmrPlatform(platformType: String) extends Platform(platformType)
+
+class OnPremPlatform extends Platform(PlatformTypes.ONPREM)
+
+/**
+ * Factory for creating instances of different platforms.
+ * This factory supports various platforms and provides methods for creating
+ * corresponding platform instances.
+ */
+object PlatformFactory extends Logging {
+  private lazy val platformInstancesMap: Map[String, Platform] = Map(
+    PlatformTypes.DATABRICKS_AWS -> new DatabricksPlatform(PlatformTypes.DATABRICKS_AWS),
```

Comment:
I'm not sure this is a big deal but whenever this map is referenced its going to do the new on all of its contents, so it creates all of these objects even though we won't use them.  Probably better off doing a match statement here and only creating the one we actually need.

Reply:
Replaced `map` with `pattern matching`